packages/dds/task-manager/src/taskManager.ts

```diff
@@ -770,7 +770,23 @@ export class TaskManager extends SharedObject<ITaskManagerEvents> implements ITa
 		}
 	}
 
-	public applyStashedOp() {
-		// do nothing...
+	protected applyStashedOp(content: any): void {
+		const taskOp: ITaskManagerOperation = content;
+		switch (taskOp.type) {
+			case "abandon": {
+				this.abandon(taskOp.taskId);
+				break;
+			}
+			case "complete": {
+				this.complete(taskOp.taskId);
+				break;
+			}
+			case "volunteer": {
+				this.subscribeToTask(taskOp.taskId);
```

Comment:
Actually this one should probably just be
```suggestion
				this.volunteerForTask(taskOp.taskId);
```

Reply:
that method is async, and can't be called in this context. subscribeToTask seems to do that same thing

Reply:
Task subscription is different from a one-time volunteer - as you have it here, the client will immediately try to volunteer again after completing connection.  This will likely cause zombie task holders as the client code won't be set up to receive a task without having invoked the call itself.

Neither is perfect since we can't deduce which method introduced the volunteer op into the stream (it's a runtime choice), but it's generally going to be safer to avoid volunteering implicitly upon connect.

You can call it as `this.volunteerForTask(...).catch(() => { })` -- the returned promise is just syntax sugar for awaiting task acquisition and is safe to ignore/not wait for.